### PR TITLE
Adds a more step by step guide to Federation

### PR DESCRIPTION
**Pull request by darrellwarde**
Adds a more step by step guide to Federation. Also adds instructions on how to optimally create a new project.

Fixes some markdown style code blocks.

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-graphql-9.surge.sh

File: modules/ROOT/pages/how-to-guides/create-project.adoc [hunk 1]
```diff
@@ -0,0 +1,126 @@
+= How to create a new project
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/create-project.adoc:1]: I believe this is more or less the same that we have in the Getting started, though? I have worked a bit on it in this https://github.com/neo4j/docs-graphql/pull/8PR

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 2]
```diff
@@ -0,0 +1,150 @@
+[[apollo-federation]]
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:1]:
```suggestion
[[apollo-federation]]
:description: This guide shows how to create a subgraph using the Neo4j GraphQL Library, for composition into a supergraph using Apollo Federation.
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+= How to create a subgraph compatible with Apollo Federation
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:2]:
```suggestion
= Create a subgraph compatible with Apollo Federation
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+This guide shows you how to create a subgraph using the Neo4j GraphQL Library, for composition into a supergraph using Apollo Federation. In this guide, the following monolithic schema will be converted into a subgraph schema for use behind a Federation gateway:
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:4]:
```suggestion
This guide shows how to create a subgraph using the Neo4j GraphQL Library, for composition into a supergraph using Apollo Federation. 

The objective is to convert the following monolithic schema into a subgraph schema for use behind a Federation gateway:
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+const server = new ApolloServer({
+  schema,
+});
+
+const { url } = await startStandaloneServer(server);
+console.log(`🚀  Server ready at ${url}`);
+----
+
+== 1. Create a new project
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:43]:
```suggestion
== Create a new project and install the Apollo Server
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+Follow the steps in xref::how-to-guides/create-project.adoc.
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:45]:
```suggestion
. Follow the steps in xref::how-to-guides/create-project.adoc[How to create a new project].

. Besides the dependencies installed in the previous step, you must install `@apollo/server`, the library for Apollo Server.
It is used in this guide to host the subgraph:

[source, bash]
----
npm install @apollo/server
----
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+== 2. Install dependencies
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:47]: If you accept the previous suggestion, please delete this section

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+The final dependency to install is `@apollo/server`, the library for Apollo Server. It is used in this guide to host the subgraph.
+
+Run the following command to install this final package:
+
+[source, bash]
+----
+npm install @apollo/server
+----
+
+== 3. Opt in to Federation
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:60]:
```suggestion
== Opt in to Federation
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+  extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key"])
+
+  type User {
+    id: ID!
+    name: String!
+  }
+`;
+----
+
+NOTE: This example only includes the Federation `@key` directive. As you want to use more https://www.apollographql.com/docs/federation/federated-types/federated-directives[Federation directives], you will need to add them to the `import` array.
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:76]:
```suggestion
Note that this example only includes the Federation `@key` directive. 
To use more https://www.apollographql.com/docs/federation/federated-types/federated-directives[Federation directives], add them to the `import` array.
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+== 4. Define an entity
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:78]:
```suggestion
== Define an entity
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+In order for other subgraphs to be able to contribute fields to the `Movie` type, we must define that type as an https://www.apollographql.com/docs/federation/entities/[entity]. You do this by using the `@key` directive to designate a field (or fields) as a key:
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:80]:
```suggestion
Defining a type as an https://www.apollographql.com/docs/federation/entities/[entity] allows other subgraphs to contribute with fields to the `Movie` type.
To achieve that, use the `@key` directive to designate a field (or fields) as a key:
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+  extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key"])
+
+  type User @key(fields: "id") {
+    id: ID!
+    name: String!
+  }
+`;
+----
+
+NOTE: Whilst only the `@key` directive has been added in this example, you should consider using either the `@id` or `@unique` directive on the `id` field. The Federation gateway expects each key to resolve to one result, so it is good practice to ensure that these values are unique in the database.
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:94]:
```suggestion
Although only the `@key` directive has been added to this example, consider using either the `@id` or the `@unique` directives on the `id` field.
The Federation gateway expects each key to resolve to one result, so it is good practice to ensure that these values are unique in the database.
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+== 5. Generate a subgraph schema
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:96]:
```suggestion
== Generate a subgraph schema
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+
+When using the Neo4j GraphQL Library, generating the subgraph schema is as simple as calling `getSubgraphSchema` instead of `getSchema`, so only the following line needs to be changed:
+
+[source, javascript]
+----
+const schema = neo4jgraphql.getSubgraphSchema();
+----
+
+== Complete example
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:105]:
```suggestion
== Conclusion
```

File: modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+
+The above snippets are combined below:
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/apollo-federation.adoc:107]:
```suggestion
By combining all previous snippets, you should get this:
```

File: modules/ROOT/pages/how-to-guides/integrations/nextjs.adoc [hunk 16]
```diff
@@ -0,0 +1,4 @@
+= How to use the Neo4j GraphQL Library with Next.js
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/nextjs.adoc:1]:
```suggestion
[[next.js]]
:description: This page shows information on how to use the Neo4j GraphQL Library with Next.js
= Use the Neo4j GraphQL Library with Next.js
```

File: modules/ROOT/pages/how-to-guides/integrations/nextjs.adoc [hunk 17]
```diff
@@ -0,0 +1,4 @@
...
+
+We maintain an https://github.com/vercel/next.js/tree/canary/examples/with-apollo-neo4j-graphql[example project] in the Next.js repository  on how to use the Neo4j GraphQL Library with Next.js.
```
Comment (lidiazuin) [inline modules/ROOT/pages/how-to-guides/integrations/nextjs.adoc:3]:
```suggestion
This https://github.com/vercel/next.js/tree/canary/examples/with-apollo-neo4j-graphql[example project] in the Next.js repository demonstrates how to use Neo4j with https://neo4j.com/docs/getting-started/appendix/example-data/[the "Movies" example], the Neo4j GraphQL Library and https://www.apollographql.com/docs/apollo-server/[Apollo Server].
```

**Review by lidiazuin — CHANGES_REQUESTED** [review]
I'm not sure if we need the create-project.adoc if considered what we have in getting-started/index.adoc. Please check this PR https://github.com/neo4j/docs-graphql/pull/8 to make sure it's the same thing so we can make adjustments . Thanks!

**Review by lidiazuin — CHANGES_REQUESTED** [review]
I'm not sure if we need the create-project.adoc if considered what we have in getting-started/index.adoc. Please check this PR https://github.com/neo4j/docs-graphql/pull/8 to make sure it's the same thing so we can make adjustments . Thanks!

**Comment by darrellwarde** [comment]
> I'm not sure if we need the create-project.adoc if considered what we have in getting-started/index.adoc. Please check this PR #8 to make sure it's the same thing so we can make adjustments . Thanks!

My thinking was that creating a project is going to be a fairly common step in many different how-to guides, so we should maybe have a centralised how-to guide on this which all of those other guides can point to.

We can't point to the Getting Started guide because it's a much more holistic example, containing steps further than what most guides will need.

If we repeat ourselves in various different guides, they are not going to get updated so there will be discrepancies across the docs. That was my thinking anyway!

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-graphql-9.surge.sh

**Review by lidiazuin — APPROVED** [review]
LGTM! I only want to make some changes later on when turning some things into lists :) thanks for your work!

**Pull request merged**